vmm/src/vm_config.rs

```diff
@@ -440,6 +515,21 @@ pub struct DeviceConfig {
     pub x_nv_gpudirect_clique: Option<u8>,
 }
 
+impl DeviceConfig {
+    pub fn apply_landlock(self, landlock: Landlock) -> LandlockResult<Landlock> {
+        let device_flags = landlock::Perms::READ | landlock::Perms::WRITE;
+        let mut landlock = landlock;
+        //TODO: Fix this unwrap. It should be replaced with proper RulesetError
+        let device_path = fs::read_link(self.path).unwrap();
```

Comment:
@l0kod I dropped LandlockError from the previous version. The only place this is causing an issue is here.

As this check returns an `io::Error`, I cannot return a `RulesetError` from this method. I can think of below possibilities to address this:

1) re-introduce LandlockError to properly address this case
2) Extend RulesetError with another error which takes io::Error
3) forcefully shoehorn this error into one of the existing RulesetError. I checked and couldn't find anything sensible.

Reply:
Looks good. We may want to update rust-landlock but for now let's stick to this.